### PR TITLE
Gui Game Lobby

### PR DESCRIPTION
**Pull request by TimTheToad**
Local Server List with active servers found on LAN

### CODE REVIEW

File: CrazyCanvas/Source/GUI/LobbyGUI.cpp [hunk 1]
```diff
@@ -55,24 +57,50 @@ bool LobbyGUI::ConnectEvent(Noesis::BaseComponent* source, const char* event, co
...
+	newInfo.Ping = 0;
+	newInfo.LastUpdate = EngineLoop::GetTimeSinceStart();
+	newInfo.EndPoint = *event.pEndPoint;
 
-	LOG_INFO("Found server at %s with %u players", pEndPoint->ToString().c_str(), players);
+	pDecoder->ReadUInt8(newInfo.Players);
+	pDecoder->ReadString(newInfo.Name);
+	pDecoder->ReadString(newInfo.MapName);
+
+	const auto& pair = m_Servers.find(event.ServerUID);
```
Comment (TheRealChriAnd) [inline CrazyCanvas/Source/GUI/LobbyGUI.cpp]: Is this really needed?
```suggestion
```

File: Assets/NoesisGUI/Xaml/Lobby.xaml [hunk 2]
```diff
@@ -81,33 +81,84 @@
...
+                                    <Grid.RowDefinitions>
+                                        <RowDefinition/>
+                                        <RowDefinition/>
+                                        <RowDefinition/>
+                                        <RowDefinition/>
+                                        <RowDefinition/>
+                                        <RowDefinition/>
+                                        <RowDefinition/>
+                                    </Grid.RowDefinitions>
+                                    <Grid x:Name="SERVER_CONTENT_2"  Grid.Row="0">
```
Comment (frozenbanana) [inline Assets/NoesisGUI/Xaml/Lobby.xaml]: Recommended to have more  descriptive x:Name. SERVER_CONTENT_2 does not tell me what will go inside.

File: CrazyCanvas/Source/States/ServerState.cpp [hunk 3]
```diff
@@ -65,6 +66,8 @@ bool ServerState::OnServerDiscoveryPreTransmit(const LambdaEngine::ServerDiscove
 	ServerBase* pServer = event.pServer;
 
 	pEncoder->WriteUInt8(pServer->GetClientCount());
+	pEncoder->WriteString(m_ServerName);
+	pEncoder->WriteString("Bajskorv");
```
Comment (TheRealChriAnd) [inline CrazyCanvas/Source/States/ServerState.cpp]: Maybe choose a better Test string 
```suggestion
	pEncoder->WriteString("This Is The Map Name");
```

File: LambdaEngine/Include/Networking/API/IPEndPoint.h [hunk 4]
```diff
@@ -32,6 +32,9 @@ namespace LambdaEngine
 		bool operator==(const IPEndPoint& other) const;
 		bool operator!=(const IPEndPoint& other) const;
 
+
+
+
```
Comment (TheRealChriAnd) [inline LambdaEngine/Include/Networking/API/IPEndPoint.h]:
```suggestion
```

File: LambdaEngine/Include/Networking/API/IPAddress.h [hunk 5]
```diff
@@ -21,6 +21,10 @@ namespace LambdaEngine
 
 		uint64 GetHash() const;
 
+		bool operator==(const IPAddress& other) const;
+		bool operator!=(const IPAddress& other) const;
+
+
```
Comment (TheRealChriAnd) [inline LambdaEngine/Include/Networking/API/IPAddress.h]:
```suggestion
```

**Review by TheRealChriAnd — CHANGES_REQUESTED** [review]
Nice man! just a few changes

File: Assets/NoesisGUI/Xaml/Lobby.xaml (hunk 2 above)
Comment (TimTheToad) [reply Assets/NoesisGUI/Xaml/Lobby.xaml]: Absolutely, will change on both Grids!

File: CrazyCanvas/Source/States/ServerState.cpp (hunk 3 above)
Comment (TimTheToad) [reply CrazyCanvas/Source/States/ServerState.cpp]: You are correct, but you can't say that it didn't make you smile a little.

Comment (frozenbanana) [inline CrazyCanvas/Source/States/ServerState.cpp]: Bajskorvs in master. Rude!

File: CrazyCanvas/Source/GUI/SavedServerGUI.cpp [hunk 6]
```diff
@@ -10,57 +10,71 @@ using namespace LambdaEngine;
...
+	return grid;
+}
+
+Ptr<Grid> SavedServerGUI::AddServerItem(Grid* pParentGrid, const ServerInfo& serverInfo, bool isRunning)
 {
-	
 	Ptr<Grid> grid = *new Grid();
 
-	for (int i = 0; i < 4; i++)
+	for (int i = 0; i < 5; i++)
```
Comment (frozenbanana) [inline CrazyCanvas/Source/GUI/SavedServerGUI.cpp]: Could not the hardcoded loop limit be extracted from a length of something?
Not sure why 5 is the correct number.

**Review by frozenbanana — APPROVED** [review]
Sprinkled a few comment but all in all good!

File: CrazyCanvas/Source/GUI/SavedServerGUI.cpp (hunk 6 above)
Comment (TheRealChriAnd) [reply CrazyCanvas/Source/GUI/SavedServerGUI.cpp]: As you can see the index i is never used and it does therefore not iterate over anything. Therefore a constant of 5 is perfectly valid

Comment (TimTheToad) [reply CrazyCanvas/Source/GUI/SavedServerGUI.cpp]: The grid doesn't know how many children it has yet, but I'll see if there's a way to get the number of columns.

File: CrazyCanvas/Source/States/ServerState.cpp (hunk 3 above)
Comment (TimTheToad) [reply CrazyCanvas/Source/States/ServerState.cpp]: Already fixed ^^ But admit, it made you smile a little

File: CrazyCanvas/Source/GUI/LobbyGUI.cpp (hunk 1 above)
Comment (TimTheToad) [reply CrazyCanvas/Source/GUI/LobbyGUI.cpp]: No, it's really not needed.

**Review by TheRealChriAnd — APPROVED** [review]

**Pull request merged**